test/matchers/log_format.go

```diff
@@ -0,0 +1,139 @@
+package matchers
+
+import (
+	"fmt"
+	logger "github.com/ViaQ/logerr/log"
+	"github.com/onsi/gomega/types"
+	. "github.com/openshift/cluster-logging-operator/test/helpers"
+	"reflect"
+	"strings"
+)
+
+type LogMatcher struct {
+	expected interface{}
+}
+
+func FitLogFormatTemplate(expected interface{}) types.GomegaMatcher {
+	return &LogMatcher{
+		expected: expected,
+	}
+}
+
+func (m *LogMatcher) Match(actual interface{}) (success bool, err error) {
+	template, ok := m.expected.(Log)
+	if !ok {
+		return false, fmt.Errorf("matcher expects Log")
+	}
+	log, ok := actual.(Log)
+	if !ok {
+		return false, fmt.Errorf("matcher expects Log")
+	}
+
+	return CompareLog(template,log)
+}
+
+func (m *LogMatcher) FailureMessage(actual interface{}) (message string) {
+	return fmt.Sprintf("Expected\n\t%#v\nto fit \n\t%#v", actual, m.expected)
+}
+
+func (m *LogMatcher) NegatedFailureMessage(actual interface{}) (message string) {
+	return fmt.Sprintf("Expected\n\t%#v\nto not fit \n\t%#v", actual, m.expected)
+}
+
+func isNil(i interface{}) bool {
+	if i == nil {
+		return true
+	}
+	switch reflect.TypeOf(i).Kind() {
+	case reflect.Ptr, reflect.Map, reflect.Array, reflect.Chan, reflect.Slice:
+		return reflect.ValueOf(i).IsNil()
+	}
+	return false
+}
+
+func DeepFields(iface interface{}, namePrefix string) ([]reflect.Value, []string) {
+	fields := make([]reflect.Value, 0)
+	names :=  make([]string,0)
+
+	ifv := reflect.ValueOf(iface)
+	ift := reflect.TypeOf(iface)
+
+	for i := 0; i < ift.NumField(); i++ {
+		v := ifv.Field(i)
+		n := namePrefix + ifv.Type().Field(i).Name
+
+		switch v.Kind() {
+		case reflect.Struct:
+			moreFields, moreNames := DeepFields(v.Interface(), n + "_")
+			fields = append(fields, moreFields...)
+			names = append(names, moreNames...)
+		case reflect.Ptr:
+			if !isNil(v.Interface()) {
+				elm := v.Elem()
+				moreFields, moreNames := DeepFields(elm.Interface(), n + "_")
+				fields = append(fields, moreFields...)
+				names = append(names, moreNames...)
+			}
+		default:
+			fields = append(fields, v)
+			names = append(names, n)
+		}
+	}
+
+	return fields, names
+}
+
+func compareLogLogic (name string, templateValue interface{}, value interface{}) bool {
+	if templateValue == value { // Same value is ok
+		logger.V(3).Info("CompareLogLogic: Same value for", "name", name, "value", value)
+		return true
+	}
+	if templateValue == "*" && !isNil(value) { // Any value, not Nil is ok if template value is "*"
+		logger.V(3).Info("CompareLogLogic: Any value for * ", "name", name, "value", value)
+		return true
+	}
+	if strings.HasPrefix(templateValue.(string), "Pre-") { // Any prefix value is ok if template value starts with "Pre-"
```

Comment:
Would we be better served and more flexible to utilize regex and maybe do that kind of comparison if the string begins/ends with a slash like `/.foo.*/`

Reply:
@jcantrill  ... If this is ok for first version of this PR I want to stay with simple "Pre".  I totally agree that adding regex will help but for the first set of tests prefix is what we need and adding regex will complicate the tests because escape characters do not pass k8s checks and than there will be need to split the field from the template in the test, so making this a bit more complicated.

+ adding more code for regex ... so if this make sense let's keep this as-is for first version and once we will advance we will work on that area. I have more ideas for advanced checks but let's not add this at this cycle

Reply:
Update: I added support for regex and removed "Pre-".  Changed the test code to use.